crates/ruff_server/src/server.rs

```diff
@@ -252,6 +255,20 @@ impl Server {
                 },
             )),
             hover_provider: Some(types::HoverProviderCapability::Simple(true)),
+            notebook_document_sync: Some(types::OneOf::Left(NotebookDocumentSyncOptions {
+                save: Some(false),
```

Comment:
I think this should be `true`. The client should notify us when a Notebook is saved. This is true for `ruff-lsp`:

https://github.com/astral-sh/ruff-lsp/blob/b8a48009f35756dc51268beb6bbc39c21a1e3a46/ruff_lsp/server.py#L139-L139

Which then requires the server to implement the `notebookDocument/didSave` endpoint:

https://github.com/astral-sh/ruff-lsp/blob/b8a48009f35756dc51268beb6bbc39c21a1e3a46/ruff_lsp/server.py#L533-L538

Reply:
Is there a reason we should listen for `didSave` in addition to `didChange`?

Reply:
For visibility, we discussed this on Discord and in conclusion it's fine to not implement the `didSave` handler because we _always_ refresh the diagnostics on `didChange` event. And, the new server doesn't support the `lint.run` setting from `ruff-lsp`.